frontend/src/component/project/Project/ProjectSettings/ProjectApiAccess/ProjectApiAccess.tsx

```diff
@@ -33,12 +96,39 @@ export const ProjectApiAccess = () => {
 
     return (
         <div style={{ width: '100%', overflow: 'hidden' }}>
-            <ApiTokenTable
-                tokens={tokens}
-                loading={loading}
-                compact
-                filterForProject={projectId}
-            />
+            <PageContent
+                header={
+                    <PageHeader
+                        title={`API access (${rows.length})`}
+                        actions={
+                            <>
+                                <Search
+                                    initialValue={globalFilter}
+                                    onChange={setGlobalFilter}
+                                />
+                                <PageHeader.Divider />
+                                <CreateApiTokenButton
+                                    permission={CREATE_PROJECT_API_TOKEN}
+                                    path="create"
+                                />
+                            </>
+                        }
+                    />
+                }
+            >
+                <ApiTokenTable
```

Comment:
this looks like a very low level prop-drilling that's difficult to read and write

Reply:
Yeah, unfortunately improving it beyond this will take a fair bit of work due to the constraints mentioned above.

Reply:
Got it!